core/src/test/java/apoc/trigger/TriggerRestartTest.java

```diff
@@ -62,6 +74,49 @@ private void restartDb() {
         waitDbsAvailable(db, sysDb);
     }
 
+    @Test
+    public void deleteSystemNodesAtStartupIfDatabaseDoesNotExist() {
+        String dbName = "notexistent";
+
+        // create manually 2 system node, the 1st one in "neo4j" and the other in an not-existent db
+        withSystemDb(tx -> {
+            Util.mergeNode(tx, ApocTrigger, null,
+                    Pair.of(database.name(), dbName),
+                    Pair.of(name.name(), "mytest"));
+
+            Util.mergeNode(tx, SystemLabels.ApocTrigger, null,
+                    Pair.of(database.name(), db.databaseName()),
+                    Pair.of(name.name(), "mySecondTest"));
+        });
+
+        // check that the nodes have been created successfully
+        withSystemDb(tx -> {
+            Iterator<Node> dbNotExistentNodes = tx.findNodes(ApocTrigger, database.name(), dbName);
+            assertTrue( dbNotExistentNodes.hasNext() );
+
+            Iterator<Node> neo4jNodes = tx.findNodes(ApocTrigger, database.name(), db.databaseName());
+            assertTrue( neo4jNodes.hasNext() );
+        });
+
+        // check that after a restart only "neo4j" node remains
+        restartDb();
+
+        withSystemDb(tx -> {
+            Iterator<Node> dbNotExistentNodes = tx.findNodes(ApocTrigger, database.name(), dbName);
+            assertFalse( dbNotExistentNodes.hasNext() );
+
+            Iterator<Node> neo4jNodes = tx.findNodes(ApocTrigger, database.name(), db.databaseName());
+            assertTrue( neo4jNodes.hasNext() );
+        });
+    }
```

Comment:
I think we should add a test where we install a trigger for neo4j, drop neo4j and eventually the trigger nodes are wiped out

Reply:
I added it in `TriggerEnterpriseFeaturesTest.testDeleteTriggersAfterDatabaseDeletion`